projects/pino-logging-gcp-config/src/pino_gcp_config.ts

```diff
@@ -88,8 +97,37 @@ export interface GCPLoggingPinoOptions {
  */
 class GcpLoggingPino {
   serviceContext: ServiceContext | null = null;
+  traceGoogleCloudProjectId: string | null = null;
 
   constructor(options?: GCPLoggingPinoOptions) {
+    this.initializeOptionsAsync(options).then(
+      () => {
+        this.outputDiagnosticEntry();
+      },
+      () => {
+        // Ignore any errors raised by resolveOptions.
+        // Errors can occur if not running in a GCP environment.
+      }
+    );
+  }
+
+  /**
+ * Resolves and initializes the configuration options for the logger.
+ *
+ * This function sets up the `serviceContext` and `traceGoogleCloudProjectId` 
+ * properties for the logger. It uses the provided options or attempts to 
+ * auto-detect values from the environment if they are not specified.
+ *
+ * @param options Configuration options for GCP logging.
+ * 
+ * @throws {Error} If `serviceContext.service` is provided but is not a valid 
+ * string or is empty.
+ */
+  async initializeOptionsAsync(options?: GCPLoggingPinoOptions) {
+    // Initializing a Cloud Logger for if we need to retrieve the
+    // ServiceContext and project ID automatically from the environment.
+    const cloudLog = new Logging({auth: options?.auth});
```

Comment:
This instantiation of Logging is not needed if both serviceContext and traceGoogleCloudProjectId are set, and may (?) fail if not in a valid GCP environment (eg isolated testing env?)

Suggestion: change to: 
```js
const cloudLog = (options?.serviceContext && options?.traceGoogleCloudProjectId)  ? null : new Logging({auth: options?.auth});
```
(or equivalent)